### PR TITLE
*: store db ID in model.TableInfo and move SchemaByTable method out from InfoSchema

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Issue Number: ref #50959

Problem Summary:

### What changed and how does it work?

In the future, we'll not store all the table info in memory, the schema -> table mapping is not available.
So how to answer which schema a table belongs to?  we can store the db ID inside the model.TableInfo and use SchemaByID

Since we can now get the db ID from model.TableInfo easily, the SchemaByTable method become redundantly, move it out from InfoSchema interface.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [X] No need to test
  > - [X] I checked and no code files have been changed.
Tiny refactor, no functional changes.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/parser/model/model.go [hunk 1]
```diff
@@ -541,6 +541,8 @@ type TableInfo struct {
 	ExchangePartitionInfo *ExchangePartitionInfo `json:"exchange_partition_info"`
 
 	TTLInfo *TTLInfo `json:"ttl_info"`
+
+	DBID int64 `json:"-"`
```
Comment (tiancaiamao) [inline pkg/parser/model/model.go:545]: This line is the core change.

**Review by GMHDBJD — APPROVED** [review]
LGTM

File: pkg/parser/model/model.go (hunk 1 above)
Comment (lcwangchao) [inline pkg/parser/model/model.go:545]: It's better to add some tests to make sure some DDL like `rename table` will reset table's `DBID` correctly, especially for schema diff load

Comment (tiancaiamao) [reply pkg/parser/model/model.go:545]: done

File: pkg/infoschema/infoschema.go [hunk 2]
```diff
@@ -589,7 +585,6 @@ func (is *SessionTables) TableByID(id int64) (tbl table.Table, ok bool) {
 // AddTable add a table
 func (is *SessionTables) AddTable(db *model.DBInfo, tbl table.Table) error {
 	schemaTables := is.ensureSchema(db)
-
 	tblMeta := tbl.Meta()
```
Comment (lcwangchao) [inline pkg/infoschema/infoschema.go:588]: Better to add an assert here?

```
intest.Assert(db.ID == tbl.Meta().DBID)
```

**Review by lcwangchao — COMMENTED** [review]
Rest LGTM

**Review by lcwangchao — APPROVED** [review]

**Review by tangenta — APPROVED** [review]

**Review by Leavrth — APPROVED** [review]

**Pull request merged**